plugins/CoreAdminHome/OptOutManager.php

```diff
@@ -215,6 +217,46 @@ public function getOptOutView()
         return $this->view;
     }
 
+    private function optOutStyling()
+    {
+        $cssfontsize = Common::getRequestVar('fontSize', false, 'string');
+        $cssfontcolour = Common::getRequestVar('fontColor', false, 'string');
+        $cssfontfamily = Common::getRequestVar('fontFamily', false, 'string');
+        $cssbackgroundcolor = Common::getRequestVar('backgroundColor', false, 'string');
+        $cssbody = 'body { ';
+
+        $hexstrings = array(
+            'fontColor' => $cssfontcolour, 
+            'backgroundColor' => $cssbackgroundcolor
+        );
+        foreach ($hexstrings as $key => $testcase) {
+            if ($testcase && !(ctype_xdigit($testcase) && in_array(strlen($testcase),array(3,6), true))) {
+                throw new \Exception("The URL parameter $key value of '$testcase' is not valid. Expected value is for example 'ffffff' or 'fff'.\n");
+            }
+        }
+
+        if ($cssfontsize && (preg_match("/^[0-9]+[\.]?[0-9]*(px|pt|em)$/", $cssfontsize))) {
```

Comment:
I'd also support `%` and `rem` (not because they are useful in this context, but it may confuse people if their value isn't accepted).
There are far more valid values, but that should be the most common ones.
https://developer.mozilla.org/de/docs/Web/CSS/font-size
https://developer.mozilla.org/de/docs/Web/CSS/length